### PR TITLE
Add `cargo-binstall` support to `wasm-bindgen`

### PR DESCRIPTION
**Pull request by NobodyXu**
This would enable `cargo-binstall` to install the pre-built binaries from release artifacts.

How to test this:

```
cargo binstall --manifest-path crates/cli wasm-bindgen-cli
```

### CODE REVIEW

**Comment by NobodyXu** [comment]
> Is this still requires as of [cargo-bins/cargo-binstall#1245](https://github.com/cargo-bins/cargo-binstall/pull/1245)?

Yes, since the artifact name used in release is `wasm-bindgen` instead of `wasm-bindgen-cli` which is the crate-name.

**Comment by daxpedda** [comment]
I'm not familiar with `cargo-binstall`, would you mind confirming that this works?
Small nit before we merge this: could you add a note in the changelog?

**Comment by NobodyXu** [comment]
> I'm not familiar with `cargo-binstall`, would you mind confirming that this works?

I've tested locally before pushing this

> Small nit before we merge this: could you add a note in the changelog?

Sure!

**Comment by NobodyXu** [comment]
@daxpedda I've updated changelog and also README on how to install `wasm-bindgen-cli`.

File: README.md [hunk 1]
```diff
@@ -26,6 +26,22 @@
...
+```
+cargo binstall wasm-bindgen-cli
+```
+
+Otherwise, you can either download it from the
+[release](https://github.com/rustwasm/wasm-bindgen/releases) page or run:
+
+```
+cargo install wasm-bindgen-cli
+```
```
Comment (daxpedda) [inline README.md:43]: I don't feel entirely comfortable promoting a third party tool I know nothing about.
Could you change the order like this:
1. `cargo install wasm-bindgen-cli`
2. GitHub Release Artifacts
3. `cargo-binstall`

Thank you by the way, improving the documentation is sorely needed.

**Review by daxpedda — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (NobodyXu) [reply README.md]: Thanks, I've changed the order as requested.

File: README.md [hunk 2]
```diff
@@ -26,6 +26,24 @@
   <sub>Built with 🦀🕸 by <a href="https://rustwasm.github.io/">The Rust and WebAssembly Working Group</a></sub>
 </div>
 
+## Install `wasm-bindgen-cli`
+
+You can install it using `cargo-install`:
```
Comment (daxpedda) [inline README.md:31]:
```suggestion
You can install it using `cargo install`:
```
`cargo install` is not a binary crate.

File: README.md [hunk 3]
```diff
@@ -26,6 +26,24 @@
...
+
+```
+cargo install wasm-bindgen-cli
+```
+
+Or, you can either download it from the
+[release](https://github.com/rustwasm/wasm-bindgen/releases) page or run:
```
Comment (daxpedda) [inline README.md:38]:
```suggestion
Or, you can download it from the
[release page](https://github.com/rustwasm/wasm-bindgen/releases).
```

File: README.md [hunk 4]
```diff
@@ -26,6 +26,24 @@
...
+
+If you have [`cargo-binstall`](https://github.com/cargo-bins/cargo-binstall) installed,
```
Comment (daxpedda) [inline README.md:40]:
```suggestion
If you have [`cargo-binstall`](https://crates.io/crates/cargo-binstall) installed,
```

File: README.md [hunk 5]
```diff
@@ -110,4 +128,4 @@ Unless you explicitly state otherwise, any contribution intentionally submitted
 for inclusion in this project by you, as defined in the Apache-2.0 license,
 shall be dual licensed as above, without any additional terms or conditions.
 
-[contributing]: https://rustwasm.github.io/docs/wasm-bindgen/contributing/index.html
\ No newline at end of file
+[contributing]: https://rustwasm.github.io/docs/wasm-bindgen/contributing/index.html
```
Comment (daxpedda) [inline README.md:131]: Can you figure out what happened here? I assume some line ending change?

**Review by daxpedda — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (NobodyXu) [reply README.md:131]: It's a newline added to the end of the file, I think it should be ok.

Comment (daxpedda) [reply README.md:131]: :/
GitHub used to show this with an icon at the end ...

**Review by daxpedda — APPROVED** [review]
Thank you!

**Pull request merged**